imap_processing/ena_maps/utils/corrections.py

```diff
@@ -0,0 +1,299 @@
+"""L2 corrections common to multiple IMAP ENA instruments."""
+
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+
+
+class PowerLawFluxCorrector:
+    """
+    IMAP-Lo flux correction algorithm implementation.
+
+    Based on Section 5 of the Mapping Algorithm Document. Applies corrections for
+    ESA transmission integration over energy bandpass using iterative
+    predictor-corrector scheme to estimate source fluxes from observed fluxes.
+
+    Parameters
+    ----------
+    coeffs_file : str or Path
+        Location of CSV file containing ESA transmission coefficients.
+    """
+
+    def __init__(self, coeffs_file: str | Path):
+        """Initialize PowerLawFluxCorrector."""
+        self._load_coefficients(coeffs_file)
+
+    def _load_coefficients(self, coeffs_file: str | Path) -> None:
+        """
+        Load ESA transmission scale factor coefficients from CSV file.
+
+        Parameters
+        ----------
+        coeffs_file : str or Path
+            Location of CSV file with coefficients.
+        """
+        self.eta_coeffs_df = pd.read_csv(coeffs_file, index_col="esa_step")
+
+    def eta_esa(self, k: np.ndarray, gamma: np.ndarray) -> np.ndarray:
+        """
+        Calculate ESA transmission scale factor η_esa,k(γ) for each energy level.
+
+        Parameters
+        ----------
+        k : np.ndarray
+            Energy levels.
+        gamma : np.ndarray
+            Power-law slopes.
+
+        Returns
+        -------
+        np.ndarray
+            ESA transmission scale factors.
+        """
+        k = np.atleast_1d(k)
+        gamma = np.atleast_1d(gamma)
+        coeff_columns = ["M0", "M1", "M2", "M3", "M4", "M5"]
+        eta = np.empty_like(gamma)
+        for i, esa_step in enumerate(k):
+            # np.polyval expects polynomial coeffs in reverse order
+            eta[i] = np.polyval(
+                self.eta_coeffs_df.loc[esa_step, coeff_columns].values[::-1], gamma[i]
+            )
```

Comment:
Not sure if this would be better or not, but I think numpy would like to transition to the Polynomial class. So something like this to keep things in the same order might work too?
https://numpy.org/doc/stable/reference/routines.polynomials.html

```suggestion
            eta[i] = np.polynomial.Polynomial(
                self.eta_coeffs_df.loc[esa_step, coeff_columns].values)(gamma[i])
            )
```